framework/src/tinyalsa/tinyalsa.c

```diff
@@ -532,14 +532,8 @@ int pcm_readi(struct pcm *pcm, void *data, unsigned int frame_count)
 	nbytes = pcm_frames_to_bytes(pcm, frame_count);
 
 #ifdef CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS
```

Comment:
add below in above or negative tc will fail
		if (pcm == NULL || data == NULL || frame_count == 0) {
			return -EINVAL;
		}